challenge/tlsalpn01/tls_alpn_challenge_test.go

```diff
@@ -116,3 +120,72 @@ func TestChallengeInvalidPort(t *testing.T) {
 	assert.Contains(t, err.Error(), "invalid port")
 	assert.Contains(t, err.Error(), "123456")
 }
+
+func TestChallengeIPaddress(t *testing.T) {
+	_, apiURL := tester.SetupFakeAPI(t)
+
+	domain := "127.0.0.1"
+	port := "23457"
+
+	mockValidate := func(_ *api.Core, _ string, chlng acme.Challenge) error {
+		conn, err := tls.Dial("tcp", net.JoinHostPort(domain, port), &tls.Config{
+			ServerName:         reverseaddr(domain),
```

Comment:
```suggestion
		arpa, _ := dns.ReversAddr(domain)
		conn, err := tls.Dial("tcp", net.JoinHostPort(domain, port), &tls.Config{
			ServerName:         arpa,
```

And add `github.com/miekg/dns` to the `import`s.